# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1347935743 14400
#      Mon Sep 17 22:35:43 2012 -0400
# Node ID 66e17621f99a132b0c3c1d3fce1a8d8c52723742
# Parent  2e5ba861898f386c3aef44a52e83a99a805fcd93
* variables.h, variables.cc (clear_current_scope): New function.

diff --git a/libinterp/interpfcn/variables.cc b/libinterp/interpfcn/variables.cc
--- a/libinterp/interpfcn/variables.cc
+++ b/libinterp/interpfcn/variables.cc
@@ -2335,18 +2335,18 @@ without the dash as well.\n\
   int argc = args.length () + 1;
 
   string_vector argv = args.make_argv ("clear");
 
   if (! error_state)
     {
       if (argc == 1)
         {
-          do_clear_globals (argv, argc, 1);
-          do_clear_variables (argv, argc, 1);
+          do_clear_globals (argv, argc, true);
+          do_clear_variables (argv, argc, true);
         }
       else
         {
           int idx = 0;
 
           bool clear_all = false;
           bool clear_functions = false;
           bool clear_globals = false;
@@ -2455,16 +2455,22 @@ without the dash as well.\n\
                 }
             }
         }
     }
 
   return retval;
 }
 
+void
+clear_current_scope (void)
+{
+  Fclear ();
+}
+
 DEFUN (whos_line_format, args, nargout,
   "-*- texinfo -*-\n\
 @deftypefn  {Built-in Function} {@var{val} =} whos_line_format ()\n\
 @deftypefnx {Built-in Function} {@var{old_val} =} whos_line_format (@var{new_val})\n\
 @deftypefnx {Built-in Function} {} whos_line_format (@var{new_val}, \"local\")\n\
 Query or set the format string used by the command @code{whos}.\n\
 \n\
 A full format string is:\n\
diff --git a/libinterp/interpfcn/variables.h b/libinterp/interpfcn/variables.h
--- a/libinterp/interpfcn/variables.h
+++ b/libinterp/interpfcn/variables.h
@@ -143,9 +143,12 @@ extern OCTINTERP_API void munlock (const
 extern OCTINTERP_API bool mislocked (const std::string&);
 
 extern OCTINTERP_API void clear_function (const std::string& nm);
 extern OCTINTERP_API void clear_variable (const std::string& nm);
 extern OCTINTERP_API void clear_symbol (const std::string& nm);
 
 extern OCTINTERP_API void maybe_missing_function_hook (const std::string& name);
 
+// Equivalent to Fclear without any arguments.
+extern OCTINTERP_API void clear_current_scope (void);
+
 #endif
